shark-hprof/src/test/java/HeapDumpRule.kt

```diff
@@ -0,0 +1,33 @@
+import com.sun.management.HotSpotDiagnosticMXBean
+import org.junit.rules.ExternalResource
+import org.junit.rules.TemporaryFolder
+
+import java.io.File
+import java.io.IOException
+import java.lang.management.ManagementFactory
+import java.util.UUID
+
+class HeapDumpRule : ExternalResource() {
+    private val temporaryFolder = TemporaryFolder()
+
+    @Throws(Throwable::class)
+    override fun before() {
+        temporaryFolder.create()
+    }
+
+    override fun after() {
+        temporaryFolder.delete()
+    }
+
+    @Throws(IOException::class)
+    fun dumpHeap(): File {
+        val hotSpotDiag = ManagementFactory.newPlatformMXBeanProxy(
+                ManagementFactory.getPlatformMBeanServer(),
+                "com.sun.management:type=HotSpotDiagnostic",
+                HotSpotDiagnosticMXBean::class.java
+        )
+        val hprof = File(temporaryFolder.root, "heapDump" + UUID.randomUUID() + ".hprof")
```

Comment:
Also thinking about it, this code does not compile top a companion object, which may be useful when a large number of test use this.